### ISSUE TITLE
RET-4 — Retire the dipole/dielectric/polar/magnetic model classes and `extensions.py`

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1600]
**Depends on:** RET-3 (#1599), ELEC-2 (#1592), ELEC-4 (#1635), MAG-1 (#1584), INF-5 (#1554) · **Blocks:** RET-5 (#1601)

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** Legacy MACE ships **eight** special-cased model classes that subclass `torch.nn.Module` / `ScaleShiftMACE` and override `forward` to read out non-energy targets, and this ticket retires all of them. Three are dipole/dielectric models in `mace/modules/models.py` — `AtomicDipolesMACE` (`:626`), `AtomicDielectricMACE` (`:842`), `EnergyDipolesMACE` (`:1199`) — and five live in `mace/modules/extensions.py` (2105 lines): the electrostatics pair `MACELES` (`:142`) and `PolarMACE` (`:663`), and the **magnetic family** `MagneticMACE` (`:1428`), `MagneticScaleShiftMACE` (`:1706`) and `MagneticSCFMACE` (`:1968`). The same file also holds `SHModule` (`:1351`, the sphericart spherical-harmonics wrapper) and `ChebyshevBasisGeneral` (`:1374`, an ordinary radial basis); deleting the file retires those too, and their v1 successors are the native spherical harmonics and the ARCH-1 (#1559) radial bases, so nothing of theirs needs preserving here.

In v1 this whole family is replaced three different ways: dipole and polarizability become **observables realized as automatic heads** on the extra-readouts axis (ARCH-3 (#1562) — declaring the observable in config creates the head; the three dipole model classes disappear); the electrostatics models are rebuilt as first-class v1 models under `models/electrostatics.py` (ELEC-2 (#1592) for `PolarMACE`, ELEC-4 (#1635) for `MACELES` — `BaseMACE` subclasses with `MACEOutputs` + a model-transform hook, because k-space electrostatic energies depend on positions and cell and couple into the derivative engine, so they are models, not readouts); and the magnetic family becomes declarative data + observables + transforms (MAG-1 (#1584) — magmom as a per-atom input feature, the magnetic-moment observable and its `dE/dm` derivative, the SCF variant on the TRN-2 (#1576) fixed-point hook). This ticket physically removes the legacy counterparts. It is a **deletion-only** PR: zero new v1 logic, only the deletion of code the frozen legacy oracle no longer needs to serve.

**Interface & constraints:**

This is a strangle-order retirement step, not a feature. The binding contract is the retirement discipline, not a new API:

1. **Preconditions (all must hold before this PR merges).** The replacements are shipped, are the **default** engine, are through the deprecation window, and have **green parity**:
   - The extra-readouts axis is default-`v1`: dipole and polarizability observables train and evaluate as automatic heads (ARCH-3 (#1562)), so the three legacy dipole model classes have a v1 replacement in production. That includes the **MDP path** specifically: `AtomicDielectricMACE` is what `mace_mdp` loads (`mace/calculators/foundations_models.py:611`, via `MACECalculator(model_type="DipolePolarizabilityMACE")`), and `get_dielectric_derivatives` (`mace/calculators/mace.py:859`) admits only `DipoleMACE`/`DipolePolarizabilityMACE`. Both must be live in v1 before the class goes.
   - The v1 electrostatics models are delivered and default-`v1`: `PolarMACE` on the new architecture (ELEC-2 (#1592)) with all three MACE-Polar artifacts converted onto it (FM-2 (#1587)), and `MACELES` with its full BEC/latent/external-field surface (ELEC-4 (#1635)).
   - The v1 magnetic family is delivered and default-`v1` (MAG-1 (#1584)), including the two behaviours develop fixed and v1 must reproduce: `MagneticSCFMACE` **delegates attribute lookup to the model it wraps** (`extensions.py:1991-2011`, so heads/`r_max`/`num_interactions`/blocks resolve through the wrapper in eval, export and fine-tuning), and the magnetic calculator **refuses `get_hessian` for an SCF-wrapped model** with an explanatory error rather than returning a fixed-moment hessian mislabelled as the self-consistent one (`mace/calculators/mace.py:1452-1473`).
   - The five weight-conversion CLIs are already gone (RET-3 (#1599)), so no legacy weight path still reaches these classes.

2. **Deletion-only — zero new v1 logic.** This PR must not add or modify any `packages/` behaviour. It deletes legacy code and the frozen-legacy wiring that instantiates or registers it. Legacy is read in place elsewhere; here it is removed.

3. **What is deleted.** The model classes plus every legacy reference that would otherwise leave a dangling import, so that the frozen `mace/` package still imports cleanly and `--engine=legacy` simply no longer offers these models:
   - `mace/modules/extensions.py` — the whole file (`MACELES`, `PolarMACE`, `MagneticMACE`, `MagneticScaleShiftMACE`, `MagneticSCFMACE`, `SHModule`, `ChebyshevBasisGeneral`, and the `graph_longrange` import guard at `:14-24`).
   - `AtomicDipolesMACE`, `AtomicDielectricMACE`, `EnergyDipolesMACE` in `mace/modules/models.py`, and any blocks/readouts that become dead only because of these removals — including the LES readout blocks (`LinearLesReadoutBlock`, `NonLinearLesReadoutBlock`) and the magnetic interaction blocks (`MagneticRealAgnosticSpinOrbitCoupledDensityInteractionBlock` and its `Residue` sibling) if nothing surviving uses them.
   - Their registry exports in `mace/modules/__init__.py`: the `from .extensions import (MACELES, MagneticScaleShiftMACE, MagneticSCFMACE, PolarMACE)` block at `:31-36`, the magnetic interaction-block imports at `:15-16` and their `interaction_classes` entries at `:77-78`, the three dipole names in the `from .models import (...)` block, and the corresponding `__all__` entries at `:120-128`.
   - The legacy dispatch that constructs them: the `mace/tools/model_script_utils.py` branches (`:91-92,117-122,131-132` the model-name allowlists and the `MACELES → FoundationMACELES` rewrite; `:258-260` magnetic; `:323-330` polar; `:367-371` and `:431-435` LES) and the `mace/tools/scripts_utils.py` per-class branches keyed on those names (`:233-239`, `:330`, `:346-353`, `:1012`, `:1047-1050`).
   - The `--model` choices in `mace/tools/arg_parser.py:137-148` for `PolarMACE`, `MACELES`, `AtomicDipolesMACE`, `AtomicDielectricMACE`, `EnergyDipolesMACE`, `MagneticScaleShiftMACE`, plus the flags that only those models consume: the 15 polar flags at `:311-395`, `--compute_polarizability`/`--compute_atomic_dipole` at `:432,438`, `--les_arguments` at `:531`, and the magnetic block at `:1152-1201` together with `--magmom_key`/`--magforces_key` (`:708,714`), `--compute_magforces` (`:444`) and the magforces weights (`:810,816-821`).
   - The calculator surfaces those classes alone serve: `MagneticMACECalculator` (`mace/calculators/mace.py:968`) and its export from `mace/calculators/__init__.py`, plus the polar/LES branches of `MACECalculator` (`:739-758`, `:798-857`, `:859`, and the `PolarMACE`/`DipoleMACE`/`DipolePolarizabilityMACE`/`EnergyDipoleMACE` `model_type` values at `:201-209`).

4. **Parity degrades to the frozen golden.** Before this PR the per-model parity tests diff the live in-process legacy oracle against v1. Because the legacy classes are being deleted, those tests can no longer run the legacy side; degrade the dipole/dielectric/polar/magnetic parity checks to compare v1 against the **frozen P0-3a and P0-3b goldens** (the committed tiny-dipole / MACE-Polar / MDP tensors from P0-3a, the tiny-magnetic tensors from P0-3b) at the same per-dtype fp64 tolerance. No tolerance is loosened. `tests/references/polar_regression_reference.json` — the legacy in-tree polar reference read at `tests/extensions/polar/test_polar_models.py:1226` — is superseded by the P0-3a golden and goes with the legacy test tree; its known-too-tight fp32 row is not carried forward.

5. **`git ls-files mace/` strictly decreases.** The retirement invariant is a monotone non-increasing legacy file count from Phase 3 onward; this PR must move it down (`mace/modules/extensions.py` entirely, plus class bodies elsewhere). The launcher's `--engine=legacy` branch itself is **not** touched here — it survives until RET-6 (#1602); this PR only shrinks what that branch can still build.

**Task:**
1. Delete `mace/modules/extensions.py` and the three dipole model classes in `mace/modules/models.py`; remove any block/readout code (LES readouts, magnetic interaction blocks) that is now unreachable solely because of these deletions.
2. Remove the registry imports, `interaction_classes` entries and `__all__` entries in `mace/modules/__init__.py` for all eight classes and the two magnetic interaction blocks.
3. Remove the legacy construction/argument wiring in `mace/tools/model_script_utils.py`, `mace/tools/arg_parser.py` and `mace/tools/scripts_utils.py`, and the polar/LES/magnetic calculator surfaces in `mace/calculators/mace.py` + `mace/calculators/__init__.py` — enough that `import mace` and the surviving `--engine=legacy` energy path still work, and no reference to a deleted symbol remains.
4. Degrade the dipole/dielectric/polar/magnetic parity tests from live-oracle diff to comparison against the frozen P0-3a / P0-3b goldens, keeping the fp64 tolerance rows unchanged.
5. Retire the now-empty extension test trees and their CI jobs: the `polar`, `polar-cueq`, `les` and `magnetic` jobs in `ci-extensions.yaml` (and their `changes` filters) lose their legacy half; the `torchsim` job's `require-caps: torchsim,polar` and the `coverage-full` job's `require-caps: network,cueq,polar,les,torchsim,schedulefree` are updated to the v1 capability set in the same PR, since a listed-but-unreachable capability passes the zero-collection guard silently.
6. Confirm nothing under `packages/` imported the deleted legacy symbols (it must not have) and that the v1 dipole/polarizability heads (ARCH-3 (#1562)), v1 electrostatics models (ELEC-2 (#1592)/ELEC-4 (#1635)) and v1 magnetic family (MAG-1 (#1584)) remain the only path.

**Out of scope:** building the v1 electrostatics models or their conversion (ELEC-2 (#1592)/ELEC-4 (#1635)); the v1 dipole/polarizability heads (ARCH-3 (#1562)); the v1 magnetic family (MAG-1 (#1584)); deleting the LAMMPS `torch.jit` path and legacy calculators (RET-5 (#1601)); deleting the `mace/tools` god-module and the launcher's legacy branch (RET-6 (#1602)); the five `convert_*` CLIs (already RET-3 (#1599)). No new v1 code of any kind.

**Acceptance criteria:**
- [ ] `mace/modules/extensions.py` no longer exists; `AtomicDipolesMACE`, `AtomicDielectricMACE`, `EnergyDipolesMACE` no longer exist in `mace/modules/models.py`.
- [ ] `import mace` and `import mace.modules` succeed with no dangling reference to a deleted symbol (`grep` for the eight class names plus `SHModule`, `ChebyshevBasisGeneral` and `MagneticMACECalculator` across `mace/` returns nothing).
- [ ] `git ls-files mace/ | wc -l` is strictly less than on the parent commit.
- [ ] The dipole/dielectric/polar/magnetic parity tests pass against the **frozen P0-3a / P0-3b goldens** at the unchanged fp64 tolerance; no tolerance edited.
- [ ] The v1 MDP path is live before deletion: `DipolePolarizabilityMACE`-equivalent loading and `get_dielectric_derivatives` both work in v1, and the latter still refuses for a polar model.
- [ ] The v1 magnetic SCF path is live before deletion: wrapper attribute delegation works, and the hessian request on an SCF-wrapped model is refused with an explanatory error rather than silently answered.
- [ ] `require-caps` lists in `ci-extensions.yaml` and `nightly.yaml` no longer name a capability whose tests were deleted here.
- [ ] The diff adds zero lines under `packages/` (deletion-only); `import-linter` still passes (`packages/` never imported these legacy symbols).
- [ ] The surviving `--engine=legacy` energy path and the v1 default path both still run end to end.

**Verify:**
```bash
test ! -e mace/modules/extensions.py
! grep -rn --include='*.py' 'AtomicDipolesMACE\|AtomicDielectricMACE\|EnergyDipolesMACE\|PolarMACE\|MACELES\|MagneticMACE\|MagneticScaleShiftMACE\|MagneticSCFMACE\|MagneticMACECalculator\|SHModule\|ChebyshevBasisGeneral' mace/
python -c "import mace, mace.modules"                                   # legacy package still imports
python -m pytest tests/parity -k "dipole or dielectric or polar or magnetic"  # now vs frozen P0-3a / P0-3b goldens (CPU, fp64)
lint-imports                                                            # import-linter: packages/ clean
git diff --stat origin/develop -- packages/ | tail -1                   # expect no additions under packages/
```
**Review focus:** confirm this is genuinely deletion-only (no v1 behaviour added or changed), that **all three** preconditions truly held — dipole/polarizability heads (including MDP + dielectric derivatives), the v1 electrostatics models, and the v1 magnetic family (including SCF attribute delegation and the hessian refusal) are default-`v1` with green parity before merge — that the frozen-golden degradation preserves every tolerance, and that `git ls-files mace/` went down while `--engine=legacy` energy training still imports and runs. Deleting the magnetic family is the easiest thing in this PR to overlook: it is in the same file, it has 31 tests and a whole CI job, and nothing in the ticket title used to say so.

---
